### PR TITLE
HTTP validator API: health endpoints

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

 Other

**What does this PR do? Why is it needed?**
migrates the following `GET` API paths to be pure rest and implements the golang `flusher` for event streams

- [x] `/v2/validator/health/version`

example of call
```
curl -X GET "http://127.0.0.1:7500/api/v2/validator/health/version" -H "accept: application/json"
{"beacon":"Prysm/Unknown/Local build. Built at: Moments ago","validator":"Prysm/Unknown/Local build. Built at: Moments ago"}
```

- [x] `/v2/validator/health/logs/validator/stream`
- [x] `/v2/validator/health/logs/beacon/stream`

screenshot of use
image

**Which issues(s) does this PR fix?**

part of #13041

### CODE REVIEW

File: validator/rpc/handlers_health.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	sub := s.logsStreamer.LogsFeed().Subscribe(ch)
+	defer func() {
+		sub.Unsubscribe()
+		close(ch)
+	}()
+	// Set up SSE response headers
+	w.Header().Set("Content-Type", "text/event-stream")
+	w.Header().Set("Cache-Control", "no-cache")
+	w.Header().Set("Connection", "keep-alive")
+	w.Header().Set("Access-Control-Allow-Origin", "*")
```
Comment (rauljordan) [inline validator/rpc/handlers_health.go:115]: Is this a potential security issue? Should be configurable, or I believe use the cors of the overall server

Comment (james-prysm) [reply validator/rpc/handlers_health.go]: removing the header for now, the cors middleware will be addressed here https://github.com/prysmaticlabs/prysm/issues/12958

**Review by rauljordan — DISMISSED** [review]

File: validator/rpc/handlers_health.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	http2.WriteJson(w, struct {
+		Beacon    string `json:"beacon"`
+		Validator string `json:"validator"`
+	}{
+		Beacon:    beacon.Version,
+		Validator: version.Version(),
+	})
+}
+
+// StreamBeaconLogs from the beacon node via a gRPC server-side stream.
```
Comment (rkapka) [inline validator/rpc/handlers_health.go:35]:
```suggestion
// StreamBeaconLogs from the beacon node via server-side events.
```
Is this more accurate?

File: validator/rpc/handlers_health.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+			return
+		default:
+			logResp, err := client.Recv()
+			if err != nil {
+				http2.HandleError(w, "could not receive beacon logs from stream", http.StatusInternalServerError)
+				return
+			}
+			jsonResp, err := json.Marshal(logResp)
+			if err != nil {
+				http2.HandleError(w, "could not encode long response", http.StatusInternalServerError)
```
Comment (rkapka) [inline validator/rpc/handlers_health.go:75]:
```suggestion
				http2.HandleError(w, "could not encode log response into JSON", http.StatusInternalServerError)
```

File: validator/rpc/handlers_health.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+	for {
+		select {
+		case log := <-ch:
+			// Set up SSE response headers
+			ls = &pb.LogsResponse{
+				Logs: []string{string(log)},
+			}
+			jsonLogs, err = json.Marshal(ls)
+			if err != nil {
+				http.Error(w, "Failed to marshal logs", http.StatusInternalServerError)
```
Comment (rkapka) [inline validator/rpc/handlers_health.go:144]: Should be `http2.HandleError` and we should wrap `err`

File: validator/rpc/handlers_health.go [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+				return
+			}
+			_, err = fmt.Fprintf(w, "%s\n", jsonLogs)
+			if err != nil {
+				http.Error(w, "Error sending data", http.StatusInternalServerError)
```
Comment (rkapka) [inline validator/rpc/handlers_health.go:149]: Should be `http2.HandleError` and we should wrap `err`

File: validator/rpc/handlers_health.go [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+				return
+			}
+
+			flusher.Flush()
+		case <-s.ctx.Done():
+			return
+		case err := <-sub.Err():
+			http.Error(w, fmt.Sprintf("Subscriber error: %v", err), http.StatusInternalServerError)
```
Comment (rkapka) [inline validator/rpc/handlers_health.go:157]: Should be `http2.HandleError`

File: validator/rpc/handlers_health.go [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+			}
+			_, err = fmt.Fprintf(w, "%s\n", jsonLogs)
+			if err != nil {
+				http.Error(w, "Error sending data", http.StatusInternalServerError)
+				return
+			}
+
+			flusher.Flush()
+		case <-s.ctx.Done():
+			return
```
Comment (rkapka) [inline validator/rpc/handlers_health.go:155]: We should return an error

File: validator/rpc/handlers_health.go [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+		case err := <-sub.Err():
+			http.Error(w, fmt.Sprintf("Subscriber error: %v", err), http.StatusInternalServerError)
+			return
+		case <-ctx.Done():
+			return
```
Comment (rkapka) [inline validator/rpc/handlers_health.go:160]: We should return an error

File: validator/rpc/handlers_health.go [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+	http2 "github.com/prysmaticlabs/prysm/v4/network/http"
+	pb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v4/runtime/version"
+	"go.opencensus.io/trace"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+// GetVersion returns the beacon node and validator client versions
+func (s *Server) GetVersion(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.web.GetVersion")
```
Comment (rkapka) [inline validator/rpc/handlers_health.go:17]:
```suggestion
	ctx, span := trace.StartSpan(r.Context(), "validator.web.health..GetVersion")
```

File: validator/rpc/handlers_health.go (hunk 8 above)
Comment (james-prysm) [reply validator/rpc/handlers_health.go:160]: it complains that you are overriding the header if we do it on context done

**Review by rkapka — APPROVED** [review]

**Pull request merged**